addons/minio/template/base/install.sh

```diff
@@ -10,10 +10,290 @@ function minio_pre_init() {
     fi
 }
 
+# ask_user_hostpath_for_migration asks uses for a path to be used during the minio host path migration.
+# this path can't be the same or a subdirectory of MINIO_HOSTPATH.
+function ask_user_hostpath_for_migration() {
+    printf "${YELLOW}\n"
+    printf "The Minio deployment is using a host path mount (volume from the node mounted inside the pod). For the\n"
+    printf "migration to proceed you have to provide the installer with a temporary directory path, this path will be\n"
+    printf "used only during the migration and will be freed after.\n"
+    printf "${NC}\n"
+
+    while true; do
+        printf "Temporary migration directory path: "
+        prompt
+
+        migration_path=$(realpath "$PROMPT_RESULT")
```

Comment:
```suggestion
        local migration_path
        migration_path=$(realpath "$PROMPT_RESULT")
```

In general I think we should try to follow the google shell style guide.

They prefer local's within functions to avoid polluting the global namespace.

https://google.github.io/styleguide/shellguide.html#s7.6-use-local-variables